pandas/core/tools/datetimes.py

```diff
@@ -663,7 +668,7 @@ def to_datetime(
 
 @overload
 def to_datetime(
-    arg: list | tuple | np.ndarray,
+    arg: list | tuple | np.ndarray | Index | ExtensionArray,
```

Comment:
Sequence?

Reply:
Could replace `np.ndarray | ExtensionArray` with `ArrayLike`.

about `Sequence` and `str`: Does to_datetime require certain list/tuple elements? I'm sure that `str` isn't part of `Sequence[datetime.time]`

Reply:
`to_datetime` can take mixed lists/tuples of different kinds of scalars (datetimes, ints, floats, strings at a minimum), and I'm a bit hesitant to figure them all out to avoid making this too narrow.  So I'm leaving this as `list | tuple` for now.